R/aum.r

```diff
@@ -0,0 +1,36 @@
+Proposed_AUM_micro<-function(pred_tensor,label_tensor){
+  n_class=pred_tensor$size(2)
+  one_hot_labels = torch::nnf_one_hot(label_tensor, num_classes=n_class) 
+  is_positive = one_hot_labels
+  is_negative =1-one_hot_labels
+  fn_diff = -is_positive$flatten()
+  fp_diff = is_negative$flatten()
+  thresh_tensor = -pred_tensor$flatten()
+  fn_denom = is_positive$sum()
+  fp_denom = is_negative$sum()
+  sorted_indices = torch::torch_argsort(thresh_tensor)
+  sorted_fp_cum = fp_diff[sorted_indices]$cumsum(dim=1) / fp_denom
+  sorted_fn_cum = -fn_diff[sorted_indices]$flip(1)$cumsum(dim=1)$flip(1) / fn_denom
+  
```

Comment:
I suggest avoiding empty lines in the middle of functions.
Please delete or add a comment explaining the different sections.